io/slow5/slow5.go

```diff
@@ -196,7 +198,9 @@ func NewParser(r io.Reader, maxLineSize int) (*Parser, []Header, error) {
 	return parser, headers, nil
 }
 
-// ParseNext parses the next read from a parser.
+// ParseNext parses the next read from a parser. Note: you should check
+// Read.Error for errors that happen within a read, while the overall error
+// for errors that happen in the parser.
```

Comment:
Actually, I don't think I agree with this behavior regardless. Why do we require clients to check two separate places for errors? It would make sense to put errors in the Read.Error field if we were parsing multiple reads at a time and wanted to not fail if there is an individual bad read, but we are reading one at a time here. 

IMO it would be best to also return read errors in the `error` return value to ensure the client notices. Thoughts?

Reply:
@TimothyStiles as well for visibility